yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -1289,3 +1289,78 @@ func (yb *TargetYugabyteDB) GetImportedSnapshotRowCountForTable(tableName string
 	log.Infof("total row count for snapshot import of table %s: %d", tableName, snapshotRowCount)
 	return snapshotRowCount, nil
 }
+
+func (yb *TargetYugabyteDB) GetGeneratedAlwaysAsIdentityColumnNamesForTable(table string) ([]string, error) {
+	schema := yb.getTargetSchemaName(table)
+	// TODO: handle case-sensitivity correctly
+	if utils.IsQuotedString(table) {
+		table = table[1 : len(table)-1]
+	} else {
+		table = strings.ToLower(table)
+	}
+	query := fmt.Sprintf(`SELECT column_name FROM information_schema.columns where table_schema='%s' AND
+		table_name='%s' AND is_identity='YES' AND identity_generation='ALWAYS'`, schema, table)
+	var identityColumns []string
+	err := yb.connPool.WithConn(func(conn *pgx.Conn) (bool, error) {
+		rows, err := conn.Query(context.Background(), query)
+		if err != nil {
+			log.Errorf("querying identity columns: %v", err)
+			return false, fmt.Errorf("querying identity columns: %w", err)
+		}
+		defer rows.Close()
+		for rows.Next() {
+			var colName string
+			err = rows.Scan(&colName)
+			if err != nil {
+				log.Errorf("scanning row for identity column name: %v", err)
+				return false, fmt.Errorf("scanning row for identity column name: %w", err)
+			}
+			identityColumns = append(identityColumns, colName)
+		}
+		return false, nil
+	})
+	return identityColumns, err
+}
+
+func (yb *TargetYugabyteDB) DisableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error {
+	log.Infof("disabling generated always as identity columns")
+	return yb.alterColumns(tableColumnsMap, "SET GENERATED BY DEFAULT")
+}
+
+func (yb *TargetYugabyteDB) EnableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error {
+	log.Infof("enabling generated always as identity columns")
+	// YB automatically resumes the value for further inserts due to sequence attached
+	return yb.alterColumns(tableColumnsMap, "SET GENERATED ALWAYS")
+}
+
+func (yb *TargetYugabyteDB) alterColumns(tableColumnsMap map[string][]string, alterAction string) error {
+	log.Infof("altering columns for action %s", alterAction)
+	for table, columns := range tableColumnsMap {
+		qualifiedTableName := yb.qualifyTableName(table)
+		batch := pgx.Batch{}
+		for _, column := range columns {
+			query := fmt.Sprintf(`ALTER TABLE %s ALTER COLUMN %s %s`, qualifiedTableName, column, alterAction)
+			batch.Queue(query)
+		}
+
+		err := yb.connPool.WithConn(func(conn *pgx.Conn) (bool, error) {
+			br := conn.SendBatch(context.Background(), &batch)
+			for i := 0; i < batch.Len(); i++ {
+				_, err := br.Exec()
+				if err != nil {
+					log.Errorf("executing query to alter columns for table(%s): %v", qualifiedTableName, err)
+					return false, fmt.Errorf("executing query to alter columns for table(%s): %w", qualifiedTableName, err)
```

Comment:
This doesn't tell for which column the error occurred. Is there any way to find that out?

Reply:
we decided to defer this for now as per discussion